fetch/fetch.go

```diff
@@ -252,14 +201,49 @@ func NewFetch(cdb *datastore.CachedDB, msh meshProvider, b system.BeaconGetter,
 	return f
 }
 
+type dataValidators struct {
+	atx         AtxValidator
+	poet        PoetValidator
+	ballot      BallotValidator
+	block       BlockValidator
+	proposal    ProposalValidator
+	tx          TxValidator
+	malfeasance MalfeasanceValidator
+}
+
+// SetValidators sets the handlers to validate various mesh data fetched from peers.
+func (f *Fetch) SetValidators(
+	atx AtxValidator,
+	poet PoetValidator,
+	ballot BallotValidator,
+	block BlockValidator,
+	prop ProposalValidator,
+	tx TxValidator,
+	mal MalfeasanceValidator,
+) {
+	f.validators = &dataValidators{
+		atx:         atx,
+		poet:        poet,
+		ballot:      ballot,
+		block:       block,
+		proposal:    prop,
+		tx:          tx,
+		malfeasance: mal,
+	}
+}
+
 // Start starts handling fetch requests.
-func (f *Fetch) Start() {
+func (f *Fetch) Start() error {
+	if f.validators == nil {
+		return errValidatorsNotSet
+	}
```

Comment:
`f.validators == nil` means a bug in the code. Perhaps instead of checking this at runtime, better prevent it altogether by requiring the user to pass all validators during the construction of `Fetch`? Similar to what was in the original code but actually to require it rather than making it an option.

Reply:
the dependency in node.go is messed up. one needs to wait for all the handlers to be initialized before setting them in fetcher. but fetcher is needed earlier than one can finish initializing all handlers...

Reply:
Then perhaps handlers should be "registered" as they are booted up?